Update developer-driven evolution doc with LRO grow-up guidance (#28973)
Azure/azure-sdk-for-net: doc/DataPlaneCodeGeneration/DeveloperDrivenEvolution.md

To add a convenience API that takes and returns strongly-typed input and output models instead of raw JSON, steps are:
1. [Create the models](#create-the-models)
2. [Create the convenience method with new signature](#create-the-convenience-method-with-new-signature)

> - [Check if convenience method needs](#check-if-convenience-method-needs)

Reviewer: I'd like to make this step a little more subtle, since in most cases a convenience method will be needed.

public virtual async Task<Response> CreateMetricFeedbackAsync(RequestContent content, RequestContext context = null);
```
You should do the following steps to create the convenience method signature:

> ### Check if convenience method needs

Reviewer 2: Do you mean  'Decide if a convenience method is needed' ?

> ### Check if convenience method needs

Reviewer 2: Do you mean  'Decide if a convenience method is needed' ?

> You don't need to add a convenience method if there is no model returned and no model as an input. Under this condition, you could skip the following steps.

Reviewer 2: ```suggestion
The convenience method is not needed if the generated method does not contain models for input and output. You could skip the following steps.
```
```suggestion
You don't need to add a convenience method if there is no model returned and no model as an input. Under this condition, you could skip the following steps.
```

> You don't need to add a convenience method if there is no model returned and no model as an input. Under this condition, you could skip the following steps.

Reviewer 2: ```suggestion
The convenience method is not needed if the generated method does not contain models for input and output. You could skip the following steps.
```
```suggestion
You don't need to add a convenience method if there is no model returned and no model as an input. Under this condition, you could skip the following steps.
```

    }
}
```

### Implement LRO method
#### Case: With RequestContent but no return value
**Generated code before (Generated/DeviceManagementClient.cs):**
```C#
namespace Azure.IoT.DeviceUpdate
{
    public partial class DeviceManagementClient
    {
        public virtual Operation ImportDevices(WaitUntil waitUntil, string action, RequestContent content, RequestContext context = null)
        {

>             using var scope = ClientDiagnostics.CreateScope("DeviceManagementClient.ImportDevices");

Reviewer 2: Only one line in the method body?

Reviewer 2: Better to indicate here that we skip the  implementaion detail.

Reviewer 2: Better to indicate here that we skip the  implementaion detail.

>             using var scope = ClientDiagnostics.CreateScope("DeviceManagementClient.ImportDevices");

Reviewer 2: Only one line in the method body?

        }
    }
}
```
**Improve the LRO method ({Other manual folder}/DeviceManagementClient.cs):**
```C#
namespace Azure.IoT.DeviceUpdate
{
    public partial class DeviceManagementClient
    {
        // The return type is still Operation because no response value

>         public virtual Operation ImportDevices(WaitUntil waitUntil, string action, ImportType type, CancellationToken cancellationToken = default)

Reviewer 2: What is 'ImportType', a model?  Do i need to implement a function called 'ImportType.ToRequestContent'?

>         public virtual Operation ImportDevices(WaitUntil waitUntil, string action, ImportType type, CancellationToken cancellationToken = default)

Reviewer 2: What is 'ImportType', a model?  Do i need to implement a function called 'ImportType.ToRequestContent'?

        {
            // We don't open a new scope here because it is just a wrapper
            // Convert model to binary content
            RequestContent requestContent = ImportType.ToRequestContent(type);

            // Call protocol method and return the operation
            return await ImportDevices(waitUntil, action, requestContent, new RequestContext() { CancellationToken = cancellationToken }).ConfigureAwait(false);
        }
    }
}
```
#### Case: Without RequestContent and return value
There is no model returned and no model as an input. Therefore, don't add a convenience method as a wrapper.
#### Case: With a return value
**Generated code before (Generated/DeviceManagementClient.cs):**
```C#
namespace Azure.IoT.DeviceUpdate
{
    public partial class DeviceManagementClient
    {

>         public virtual Operation<Response> ImportDevices(WaitUntil waitUntil, string action, RequestContent content, RequestContext context = null)

Reviewer: `Operation<Response>` is not a valid return type -- it would be `Operation<BinaryData>` for a protocol method.

        {
            using var scope = ClientDiagnostics.CreateScope("DeviceManagementClient.ImportDevices");
        }
    }
}
```
**Improve the LRO method ({Other manual folder}/DeviceManagementClient.cs):**
```C#
namespace Azure.IoT.DeviceUpdate
{
    public partial class DeviceManagementClient
    {
        // Can use same method name here
        public virtual Operation<OutputType> ImportDevices(WaitUntil waitUntil, string action, ImportType type, CancellationToken cancellationToken = default)
        {
            // We don't open a new scope here because it is just a wrapper
            // Convert model to binary content
            RequestContent requestContent = ImportType.ToRequestContent(type);

            // Call protocol method
            var lro = await ImportDevices(waitUntil, action, requestContent, new RequestContext() { CancellationToken = cancellationToken }).ConfigureAwait(false);

            // Convert Operation<BinaryData> to Operation<OutputType>

>             return ProtocolOperationHelpers.Convert(lro, r => (OutputType)r, ClientDiagnostics, "DeviceManagementClient.ImportDevices");

Reviewer 2: Do we provide the method of ProtocolOperationHelpers.Convert() ?

>             return ProtocolOperationHelpers.Convert(lro, r => (OutputType)r, ClientDiagnostics, "DeviceManagementClient.ImportDevices");

Reviewer 2: Do we provide the method of ProtocolOperationHelpers.Convert() ?
